src/Object/Transform.cpp

```diff
@@ -54,21 +54,31 @@ Transform* Transform::GetRoot() const
     return pTransform;
 }
 
-bool Transform::IsChildOf(const Transform* parent) const
+bool Transform::IsChildOf(const Transform* parent, bool recursive /* = false */) const
 {
+    if (parent == nullptr)
+    {
+        LOG_STREAM(ERROR) << "Cannot check if this Transform is a child of a nullptr. "
+                          << "This operation will return false." << std::endl;
```

Comment:
How about 

> If `parent` is `null`, `Transform::IsChildOf(nullptr)` checks if this `Transform` is a root transform

Do you think it adds flexibility to the API or reduces readability/clarity/maintainability/whatever?

Reply:
I am vacillating between allowing this or not.

If we are pretty sure that there will never be something like a "fake object" representing a "scene when it comes to a multiple-scene-loaded scenario, we can safely allow `Transform::IsChildOf(nullptr)` to act as a root transform check

Reply:
I don't think that's going to be the case.

I think that there are two commonly adopted patterns that handle this. One is to explicitly create a concept of Scene with Scenes containing Objects, and the other one is to not differentiate Scenes from Objects, which, in this case, I would call these Nodes. One can argue which one is superior but personally I'm OK with either.

The best example of the latter approach is [Godot](https://docs.godotengine.org/en/stable/getting_started/step_by_step/nodes_and_scenes.html). In Godot, one creates Nodes, and Nodes can contain other Nodes. A Godot Scene is just a special type of Node that contains other Nodes, and thus you can nest Scenes within Scenes and it can work like a Unity Prefab.

The thing is, even if both approaches are acceptable, they are not interchangeable/mixable. Unity DOTS Subscenes are actually GameObjects yes, but we all know why it has to come down to this and they suffer from it. If we ever do "fake objects", then we are doing things horribly wrong, as it means that we are not clear with our abstraction level.

That said, whether to allow `IsChildOf(null)` or not is completely fine, but if not we still need a way to check if some transform is root.

Reply:
Ok

> One is to explicitly create a concept of Scene with Scenes containing Objects, and the other one is to not differentiate Scenes from Objects

Currently we are doing the former one:
Let's say we allow `IsChildOf(nullptr, recursive=false)`  to check if it's a root transform, but also provide an `IsRootTransform() const` which simply forwards a call to `IsChildOf(nullptr, false)`.

I know there might be an overhead if the optimization is not smart enough, but I would say let's make API intact first.

Reply:
> IsRootTransform() const which simply forwards a call to IsChildOf(nullptr, false)

Wait it doesn't need to be this complex we only need to return `m_Parent == nullptr` (thanks to copilot

Reply:
I'm not a fan of treating things like a function call overhead too seriously. If it can be easily prevented sure that's great, but if not it will not be my nightmare at least.

As long as we have a way to check the "rootness" I think we are good.

Reply:
Now it does:
- Log warning if someone calls to `IsChildOf(nullptr, false)` and recommend them to use `IsRootTransform()`
- Also log warning if `IsChildOf(nullptr, true)` is called, since it will always return true

Reply:
How about `IsRoot`? We have `IsChildOf` instead of `IsChildTransformOf` and `GetRoot` instead of `GetRootTransform`.